cli: TestUnavailableZip failed

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12697383?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12697383?buildTab=artifacts#/) on master @ [d1ddc072a8c10c7568166af9495e6418df3e5a22](https://github.com/cockroachdb/cockroach/commits/d1ddc072a8c10c7568166af9495e6418df3e5a22):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip2296066369
[debug --host=127.0.0.1:43511 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
[debug --host=127.154.161.229:34863 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    zip_test.go:479: panic: runtime error: invalid memory address or nil pointer dereference
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip2296066369
--- FAIL: TestUnavailableZip (73.28s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-inf-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33538

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12712643?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12712643?buildTab=artifacts#/) on master @ [1c90978d5eea42eb9d3944b1d25b409bc50a1f0b](https://github.com/cockroachdb/cockroach/commits/1c90978d5eea42eb9d3944b1d25b409bc50a1f0b):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip883658209
[debug --host=127.0.0.1:42149 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
[debug --host=127.135.237.25:45437 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    zip_test.go:479: panic: runtime error: invalid memory address or nil pointer dereference
test logs left over in: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip883658209
    panic.go:523: -- test log scope end --
--- FAIL: TestUnavailableZip (73.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12821046?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12821046?buildTab=artifacts#/) on master @ [5774ee0f6bf004d3fd22a08dcb1803a3026205c0](https://github.com/cockroachdb/cockroach/commits/5774ee0f6bf004d3fd22a08dcb1803a3026205c0):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip1009473352
[debug --host=127.0.0.1:38537 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
[debug --host=127.189.159.10:41403 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    zip_test.go:479: panic: runtime error: invalid memory address or nil pointer dereference
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip1009473352
--- FAIL: TestUnavailableZip (74.40s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -432,62 +432,68 @@ func TestUnavailableZip(t *testing.T) {
 		"debug zip --concurrency=1 --cpu-profile-duration=0 " + os.
 			DevNull + " --timeout=.5s"
 
-	c := TestCLI{
-		t:        t,
-		Server:   tc.Server(0),
-		Insecure: true,
-	}
-	defer func(prevStderr *os.File) { stderr = prevStderr }(stderr)
-	stderr = os.Stdout
-
-	out, err := c.RunWithCapture(debugZipCommand)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	// Assert debug zip output for cluster, node 1, node 2, node 3.
-	assert.NotEmpty(t, out)
-	clusterOut := []string{
-		"[cluster] requesting nodes... received response...",
-		"[cluster] requesting liveness... received response...",
-	}
-	expectedOut := clusterOut
-	for i := 1; i < tc.NumServers()+1; i++ {
-		nodeOut := baseZipOutput(i)
-
-		expectedOut = append(expectedOut, nodeOut...)
-
-		// If the request to nodes failed, we can't expect the remaining
-		// nodes to be present in the debug zip output.
-		if i == 1 && strings.Contains(out,
-			"[cluster] requesting nodes: last request failed") {
-			break
+	t.Run("server 1", func(t *testing.T) {
+		c := TestCLI{
+			t:        t,
+			Server:   tc.Server(0),
+			Insecure: true,
 		}
-	}
 
-	containsAssert(t, out, expectedOut)
+		out, err := c.RunWithCapture(debugZipCommand)
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	// Run debug zip against node 2.
-	c = TestCLI{
-		t:        t,
-		Server:   tc.Server(1),
-		Insecure: true,
-	}
+		// Assert debug zip output for cluster, node 1, node 2, node 3.
+		assert.NotEmpty(t, out)
+		clusterOut := []string{
+			"[cluster] requesting nodes... received response...",
+			"[cluster] requesting liveness... received response...",
+		}
+		expectedOut := clusterOut
+		for i := 1; i < tc.NumServers()+1; i++ {
+			nodeOut := baseZipOutput(i)
 
-	out, err = c.RunWithCapture(debugZipCommand)
-	if err != nil {
-		t.Fatal(err)
-	}
+			expectedOut = append(expectedOut, nodeOut...)
 
-	// Assert debug zip output for cluster, node 2.
-	assert.NotEmpty(t, out)
-	assert.NotContains(t, out, "[node 1]")
-	assert.NotContains(t, out, "[node 3]")
+			// If the request to nodes failed, we can't expect the remaining
+			// nodes to be present in the debug zip output.
+			if i == 1 && strings.Contains(out,
+				"[cluster] requesting nodes: last request failed") {
+				break
+			}
+		}
 
-	nodeOut := baseZipOutput(2)
-	expectedOut = append(clusterOut, nodeOut...)
+		containsAssert(t, out, expectedOut)
+	})
 
-	containsAssert(t, out, expectedOut)
+	t.Run("server 2", func(t *testing.T) {
+		// Run debug zip against node 2.
+		c := TestCLI{
+			t:        t,
+			Server:   tc.Server(1),
+			Insecure: true,
+		}
+
+		out, err := c.RunWithCapture(debugZipCommand)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Assert debug zip output for cluster, node 2.
+		assert.NotEmpty(t, out)
+		assert.NotContains(t, out, "[node 1]")
+		assert.NotContains(t, out, "[node 3]")
+
+		clusterOut := []string{
+			"[cluster] requesting nodes... received response...",
+			"[cluster] requesting liveness... received response...",
+		}
+		nodeOut := baseZipOutput(2)
+		expectedOut := append(clusterOut, nodeOut...)
+
+		containsAssert(t, out, expectedOut)
+	})
 }
 
 func containsAssert(t *testing.T, actual string, expected []string) {

